cJSON.c

```diff
@@ -882,10 +889,22 @@ char *cJSON_PrintBuffered(const cJSON *item, int prebuffer, cjbool fmt)
     }
     p.length = prebuffer;
     p.offset = 0;
+    p.noalloc = 0;
 
     return print_value(item, 0, fmt, &p);
 }
 
+int cJSON_PrintPreallocated(cJSON *item,char *buf, const size_t len, const cjbool fmt)
```

Comment:
About the `size_t`. I agree that `size_t` should be used everywhere instead of `int`, and it will be in v2, but currently `ensure` and `printbuffer` and all the other internal functions use `int` for sizes, so using `size_t` can create the illusion that numbers exceeding the integer range are supported, but they're not.

So please change to `int` 😢.